### ISSUE TITLE
Aberrant behaviour of self-sizing `frame`/`key' combination (in 7.4.11.2266)

### ISSUE THREAD

**Issue by JSearUK** [renpy/renpy#3842]
NOTE: I've not tested this in later versions; the issue may have already been resolved.

Issue: Adding a `key` action to a self-sizing `frame` maximizes it. It *does* still respect things like `maximum`, though.

Demo:
```py
default add_key = False

screen test():
    frame:
        text "Click me!"
        if add_key:
            key "mouseup_1" action Hide("test")

label start:
    show screen test
    "This is with `key` disabled (click advances dialogue, as normal):\n- frame self-sizing works as expected."
    $ add_key = True
    "This is with `key` enabled (click hides test screen):\n- frame is maxed."
    $ add_key = False
    jump start
```

**Comment by JSearUK** [comment]
> You should test it on the latest stable versions and/or nightly before reporting.

Agreed, in principle. However, I'm leery of updating to the latest versions because they are still so new and likely haven't had all the wrinkles ironed out yet. That said, I've been considering updating to 7.5.x. But last time I tried to run two different SDK's side-by-side, they both got borked. Clearly, I did something wrong, so I've not risked it since. I'm hoping that this will be less of a problem for you guys.
